.github/workflows/lint.yaml

```diff
@@ -0,0 +1,21 @@
+name: lint
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+  workflow_dispatch:
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Install dependencies
+        run: npm install
```

Comment:
https://docs.npmjs.com/cli/v8/commands/npm-ci might be safer here too. so we don't have accidental dep upgrades that break things in odd ways compared to local